src/treestatus/api/treestatus_api/api.py

```diff
@@ -36,7 +51,184 @@ def _now():
     return datetime.datetime.utcnow().replace(tzinfo=pytz.UTC)
 
 
+def _statuspage_data(
+    close,
+    component_id,
+    tree,
+    status_from,
+    status_to,
+):
+    data = {
+        'status': close and 'resolved' or 'investigating',
+        'components': {
+            component_id: close and 'operational' or 'major_outage',
+        }
+    }
+    if not close:
+        data['name'] = f'Tree {tree.tree} closed'
+        data['component_ids'] = [component_id]
+        data['metadata'] = {
+            'treestatus': {
+                'tree': tree.tree,
+                'status_from': status_from,
+                'status_to': status_to,
+            },
+        }
+        data['body'] = (
+            f'Message of the day: {tree.message_of_the_day}\n'
+            f'Reason: {tree.reason}\n'
+        )
+    return data
+
+
+def _statuspage_send_email_on_error(subject, content, incident_id=None):
+    page_id = flask.current_app.config.get('STATUSPAGE_PAGE_ID')
+    address = flask.current_app.config.get('STATUSPAGE_NOTIFY_ON_ERROR')
+
+    if not address:
+        return
+
+    link = {
+        'href': f'https://manage.statuspage.io/pages/{page_id}',
+        'text': 'Visit statuspage',
+    }
+    if incident_id:
+        link = {
+            'href': f'https://manage.statuspage.io/pages/{page_id}/incidents/{incident_id}',
+            'text': 'Visit statuspage incident',
+        }
+    flask.current_app.notify.email({
+        'address': address,
+        'subject': subject,
+        'content': content,
+        'link': link,
+    })
+
+
+def _create_statuspage_incident(
+    headers,
+    component_id,
+    tree,
+    status_from,
+    status_to,
+):
+    page_id = flask.current_app.config.get('STATUSPAGE_PAGE_ID')
+    response = requests.post(
+        f'{STATUSPAGE_URL}/pages/{page_id}/incidents',
+        headers=headers,
+        json=_statuspage_data(False,
+                              component_id,
+                              tree,
+                              status_from,
+                              status_to,
+                              ),
+    )
+    try:
+        response.raise_for_status()
+    except Exception:
+        _statuspage_send_email_on_error(
+            subject=f'[treestatus] Error when creating statuspage incident',
+            content=STATUSPAGE_ERROR_ON_CREATE.format(tree=tree.tree),
+            link=dict(
+                href=f'https://manage.statuspage.io/pages/{page_id}',
+                text='Visit statuspage',
+            ),
+        )
+        pass
+
+    return response.json()
```

Comment:
Hmm, I think you may hit an issue trying to return this in case you hit an exception in `response.raise_for_status()`. You catch the exceptions, send an email, but proceed here in any case. I'd say move this just after `response.raise_for_status()` and return an error or something here.